pycoal/mineral.py

```diff
@@ -10,225 +10,79 @@
 
-import spectral
-import pickle
 import numpy
+import spectral
 
 class MineralClassification:
-    def __init__(self):
-        # define wavelength range in micrometers
-        self.wavelength_min = 0.38315
-        self.wavelength_max = 2.5082
-
-        # define number of layers in normalized pixel
-        self.num_layers = 128
-
-    def normalize(self, image, wavelengthList=None):
-        """
-        Return a copy of an MxNxB image (or 1x1xB pixel) with constrained wavelength range and normalized spectra.
-        Args:
-           image (numpy.ndarray): the image or pixel to normalize
-           wavelengthList (list) [default None]: the list of wavelengths for the pixel
-        Returns:
-           normalizedImage (numpy.ndarray): the normalized image or pixel
-        """
-
-        # constrain wavelength range
-        if wavelengthList is None:
-            wavelengthList = map(float, image.metadata.get('wavelength'))
-
-        # get the indices of the minimum and maximum element
-        i = self.__indexOfGreaterThan(wavelengthList, self.wavelength_min)
-        j = self.__indexOfGreaterThan(wavelengthList, self.wavelength_max)
-
-        # constrain wavelength range
-        # slice the array from MxNxB to MxNx(j-i) for indices i,j
-        image = image[:,:,i:j]
-
-        # normalize spectra
-        principalComponents = spectral.algorithms.principal_components(image)
-        principalComponents.reduce(self.num_layers)
-        normalizedImage = principalComponents.transform(image)
 
-        # return normalized image
-        return normalizedImage
-
-    @staticmethod
-    def __indexOfGreaterThan(elements, value):
-        """
-        Return the index of the first element greater than the value.
-        Args:
-           elements (list): list of elements
-           value (float): the value to compare
-        Returns:
-           index (int): the index of the first element greater than the value
-        """
-
-        for index,element in enumerate(elements):
-            if element > value:
-                break
-        return index
-
-    def trainClassifier(self, libraryFileName, classifierType='gaussian'):
-        """
-        This takes the file name of the library to be used in the training and returns a
-        classifier
-        Args:
-           libraryFileName (str): the name of the training library
-           classifierType (str) [default 'gaussian']: the type of classifier.
-                                                      Currently, can be one of:
-                                                      perceptron, gaussian, or
-                                                      mahalanobisdistance.
-        Returns:
-           classifier (PerceptronClassifier, GaussianClassifier,
-                       or MahalanobisDistanceClassifier): trained classifier
+    def __init__(self, libraryFilename):
         """
+        Construct a new MineralClassification object with the USGS Digital
```

Comment:
done. that page links to the convolved library which is available over FTP:
ftp://ftpext.cr.usgs.gov/pub/cr/co/denver/speclab/pub/spectral.library/splib06.library/Convolved.libraries